kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -72,7 +72,7 @@ public String channelDescriptor() {
      */
     @Override
     public ByteBufferOutputStream createByteBufferOutputStream(int initialCapacity) {
-        final ByteBuf buffer = channelContext.alloc().heapBuffer(initialCapacity);
+        final ByteBuf buffer = channelContext.alloc().directBuffer(initialCapacity);
```

Comment:
I'm trying to reconcile your words:

> While we want to use DirectBuffers there can be issue with the availability of the API's used to manage them so we leave it up to Netty to determine if available by calling Allocator.ioBuffer rather than going directly to Allocator.directBuffer.

I was expecting to see `channelContext.alloc().ioBuffer(initialCapacity)`

Reply:
Thanks for catching @k-wall I screwed up a rebase before pushing up.